infra/extfunc-lambda/lambda_function.py

```diff
@@ -0,0 +1,204 @@
+from base64 import b64encode
+from json import dumps, loads, JSONDecodeError
+from re import match
+from urllib.request import urlopen, Request, HTTPBasicAuthHandler
+from urllib.parse import parse_qsl
+from urllib.error import HTTPError, URLError
+
+from vault import decrypt_if_encrypted
+from utils import parse_header_links, pick
+
+
+def parse_header_dict(value):
+    return {k: decrypt_if_encrypted(v) for k, v in parse_qsl(value)}
+
+
+def lambda_handler(event, context=None):
+    headers = event['headers']
+
+    # url, host, path
+    if 'sf-custom-url' in headers:
+        req_url = headers['sf-custom-url']
+        m = match(r'^https://([^/]+)(.*)$', req_url)
+        if m:
+            req_host, request_path = m.groups()
+        else:
+            raise RuntimeError('url must start with https://')
+    else:
+        req_host = headers['sf-custom-host']
+        req_path = headers.get('sf-custom-path', '/')
+        req_url = f'https://{req_host}{req_path}'
+
+    req_kwargs = parse_header_dict(headers.get('sf-custom-kwargs'))
+
+    req_headers = {
+        k: v.format(**req_kwargs)
+        for k, v in parse_header_dict(headers.get('sf-custom-headers', '')).items()
+    }
+    if 'sf-custom-basicauth' in headers:
+        basicauth = decrypt_if_encrypted(headers['sf-custom-basicauth'])
```

Comment:
the secret should get bundled with an audience and we should check that audience matches the domain before passing the secret off to it

Reply:
fixed above in `auth.get('host') != req_host` check